index.js

```diff
@@ -835,6 +856,89 @@ class Batch {
   }
 }
 
+class Watcher extends EventEmitter {
+  constructor (bee, range) {
+    super()
+
+    this.bee = bee
+    this.core = bee.core
+
+    this.watching = false
+    this.running = false
+    this.destroyed = false
+
+    this.range = range
+    this.latestDiff = this.bee.version
+    this.stream = null
+
+    const onappend = this._onappend.bind(this)
+    this.onappend = debounceify(onappend)
+
+    // this.on('error', noop)
+  }
+
+  async _onappend () {
+    try {
+      await this._run()
+    } catch (err) {
+      this.emit('error', err)
+      this.destroy()
+    } finally {
+      this.running = false
+    }
+  }
+
+  async _run () {
+    this.running = true
+
+    // + I think it could repeat a last call even after bee is closed, should trigger it with a test first
+    // if (this.destroyed) return
+
+    const snapshot = this.bee.snapshot()
+    this.stream = snapshot.createDiffStream(this.latestDiff, this.range)
+
+    try {
+      for await (const data of this.stream) { // eslint-disable-line
+        this.emit('change', snapshot.version, this.latestDiff)
```

Comment:
Any particular reason to not emit the actual diff? That's what I would be most interested in most of the time

Reply:
Yes, but if I give you only the first diff and skip over all others (due `break`ing the loop) then the diff is not useful

Reply:
That's why I wanted a `opts.includeDiff` but Maf spoke haha

Reply:
Ah, should have read that thread with more attention. Yeah, sounds reasonable